server/controllers/applicant/updateApplicant.js

```diff
@@ -0,0 +1,43 @@
+const Applicant = require('../../database/models/applicant');
+const validationUpdate = require('../../utils/validation/applicantValidation');
+
+const updateApplicant = (req, res) => {
+  const { body } = req;
+  const { id } = req.params;
+  if (!body) {
+    return res.status(400).json({
+      success: false,
+      error: 'You must provide a body to update',
+    });
+  }
+  validationUpdate(body).then((valid) => {
+    if (!valid) res.status(400).json({ err: 'error validation' });
+    else {
+      Applicant.findOne({ _id: req.params.id }, (err, applicant) => {
```

Comment:
No need to do this check with findOne as updateOne will return object that contains the number of matched rows an the modiefiedOne(exactly one)
The result would be something like this
```js
{ "acknowledged" : true, "matchedCount" : 1, "modifiedCount" : 1 }
```

With no match

```js
{ "acknowledged" : true, "matchedCount" : 0, "modifiedCount" : 0 }

```
The shape of the object could be alittle bit different between mongo and mongoose
Check this
https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/

https://mongoosejs.com/docs/api.html#model_Model.updateOne